iis-wf/iis-wf-report/src/main/java/eu/dnetlib/iis/wf/report/pushgateway/pusher/PushGatewayPusherCreator.java

```diff
@@ -0,0 +1,39 @@
+package eu.dnetlib.iis.wf.report.pushgateway.pusher;
+
+import io.prometheus.client.exporter.PushGateway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class PushGatewayPusherCreator implements MetricPusherCreator {
+    private static final Logger logger = LoggerFactory.getLogger(PushGatewayPusherCreator.class);
+
+    public MetricPusher create(String address) {
+        Supplier<PushGateway> pushGatewaySupplier = () -> {
+            try {
+                return new PushGateway(address);
+            } catch (Exception e) {
+                logger.error("PushGateway creation failed, address: {}.", address);
+                e.printStackTrace();
+                return null;
+            }
+        };
+        return create(pushGatewaySupplier);
+    }
+
+    public MetricPusher create(Supplier<PushGateway> pushGatewaySupplier) {
+        return (collectorRegistry, jobName) -> Optional
+                .ofNullable(pushGatewaySupplier.get())
+                .ifPresent(pushGateway -> {
+                    try {
+                        pushGateway.pushAdd(collectorRegistry, jobName);
```

Comment:
Should we be using `pushAdd`? It replaces existing metrics with the same name, but if some are missing in a report we will be left with old values. I would suggest rather using `push` here.

I also think job name should always be "iis" and "primary_import" etc. would better fit an additional grouping key named eg. `wf`.

We also need a label to distinguish runs for production, beta, etc. I think it can be a grouping key too, named `env`, like the label we use for services.

Reply:
From today's call:
-  let's use `user` not `env` as a grouping key
- don't use `wf`, group just by `user` - this way we can export all the available reports every time

Reply:
Changes
1. use `push`
1. job is always `iis`
1. added grouping key map with keys and values supplied from workflow; currently only one grouping key `user`